qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -328,8 +332,10 @@ def _obtain_swaps(self, front_layer, current_layout):
 
     def _compute_cost(self, layer, layout):
         cost = 0
+        layout_map = layout._v2p
+        dist_matrix = self.coupling_map._dist_matrix
```

Comment:
Since `coupling_map` is fixed for the lifetime of the pass, can we generate the `.distance_matrix` once, and store it as a class attribute?

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/95caec850a9ee4bf964a8f6fb1be3aafe774f971